### PR TITLE
feat: Reused landscape creation boundary options in update boundary

### LINKED ISSUE #461: Draw Landscape Boundary: Reuse boundary options for add and update
## Description
Change Update Boundary option to have the same boundary options displayed in the add landscape form

## Additional context

Figma: https://www.figma.com/file/BYn1r2xSKtap9V6pefEMiz/Draw-landscape---R1.3?node-id=918%3A947

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Added boundary creation options to update boundary page

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [x] English strings reviewed and copyedited
- [x] Strings are localized
- [x] Verified on mobile
- [x] Verified on desktop
- [x] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes #461 

### Verification steps

image

### CODE REVIEW

File: src/gis/components/Map.js [hunk 1]
```diff
@@ -32,11 +25,29 @@ L.Icon.Default.mergeOptions({
...
+
+function layerToGeoJSON(layers) {
+  const features = layers
+    .filter(layer => layer.toGeoJSON)
+    .map(layer => layer.toGeoJSON());
+  return {
+    type: 'FeatureCollection',
+    features,
+  };
+}
```
Comment (paulschreiber) [inline src/gis/components/Map.js:43]: question: Will these get reused? Do they belong in `utils`?

File: src/gis/components/Map.js [hunk 2]
```diff
@@ -95,146 +105,171 @@ const LeafletSearch = props => {
...
-    if (pinLocation && boundingBox) {
-      onPinLocationChange({
-        pinLocation,
-        boundingBox,
-      });
+    if (!newGeoJson) {
+      return;
     }
-  }, [boundingBox, pinLocation, onPinLocationChange]);
+    console.log('update geojson');
```
Comment (paulschreiber) [inline src/gis/components/Map.js:152]: issue: remove debug logging

File: src/gis/components/Map.js [hunk 3]
```diff
@@ -95,146 +105,171 @@ const LeafletSearch = props => {
...
-      dragend: () => {
-        const marker = markerRef.current;
-        if (marker != null) {
-          setPinLocation(marker.getLatLng());
-        }
+    geojsonToLayer(geojson, featureGroup, {
+      pointToLayer: (feature, latlng) => {
+        const marker = L.marker(latlng, { draggable: true });
+        marker.on('dragend', event => {
+          console.log('dragend', { event });
```
Comment (paulschreiber) [inline src/gis/components/Map.js:140]: issue: remove debug logging

File: src/gis/components/Map.js [hunk 4]
```diff
@@ -95,146 +105,171 @@ const LeafletSearch = props => {
...
+
+  const updateBounds = useCallback(
+    bbox => {
+      if (!onGeoJsonChange) {
+        return;
+      }
+      onGeoJsonChange(current => {
+        if (!current) {
+          return current;
+        }
```
Comment (paulschreiber) [inline src/gis/components/Map.js:204]: suggestion: This seems awkward. Can you return false or null (or just `return`) instead?

File: src/localization/locales/en-US.json [hunk 5]
```diff
@@ -268,7 +268,7 @@
         "form_step_info_label": "Landscape Details",
         "form_step_boundaries_options_label": "Landscape Boundary",
         "form_boundary_options_title": "Define Your Landscape’s Boundary",
-        "form_boundary_options_description": "Tell us about where your landscape is. Uploading a map is the most accurate. If you don’t have one, you can <1>create a GeoJSON file</1>, drop a pin, or define your boundary later.",
+        "form_boundary_options_description": "<0><0>Select how you’d like to proceed to display the landscape boundary</0><1>We highly recommend updating  the map  with the most accurate boundary of your landscape any time.</1><2>Don’t have a GeoJSON file? Create one by following a simple instruction.</2></0>",
```
Comment (paulschreiber) [inline src/localization/locales/en-US.json:271]:
```suggestion
        "form_boundary_options_description": "<0><0>Select how you’d like to define the landscape boundary</0><1>We recommend updating the map regularly with the most accurate boundary of your landscape.</1><2>Don’t have a GeoJSON file? Create one.</2></0>",
```

File: src/landscape/landscapeUtils.js [hunk 6]
```diff
@@ -13,19 +13,23 @@ const parseGeoJson = areaPolygon => {
 };
 
 export const getLandscapeBoundingBox = (landscape = {}) => {
```
Comment (paulschreiber) [inline src/landscape/landscapeUtils.js:15]: suggestion: add a comment explaining what this helper does and why.

File: src/landscape/landscapeService.js [hunk 7]
```diff
@@ -65,30 +65,32 @@ export const fetchLandscapeToView = (slug, currentUser) => {
...
+      ...landscape,
+      areaPolygon: landscape.areaPolygon
+        ? JSON.parse(landscape.areaPolygon)
+        : null,
+    }))
+    .then(landscape => {
+      if (landscape.areaPolygon) {
+        return landscape;
+      }
+      // Get Bounding box if no areaPolygon data
```
Comment (paulschreiber) [inline src/landscape/landscapeService.js:86]:
```suggestion
      // Get bounding box if no areaPolygon data
```

Comment (paulschreiber) [inline src/landscape/landscapeService.js:86]: suggestion: expand this comment to provide more context. In what circumstances is areaPolygon data not present?

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/gis/components/Map.js (hunk 1 above)
Comment (josebui) [reply src/gis/components/Map.js:43]: @paulschreiber Maybe, not right now, when it gets reused I think we should move it to a common place

**Review by paulschreiber — APPROVED** [review]

**Comment by paulschreiber** [comment]
@josebui `landscape.boundaries_update_save` needs translation

**Pull request merged**